lib/metasploit/framework/login_scanner/teamcity.rb

```diff
@@ -134,9 +134,26 @@ class TeamCityError < StandardError; end
         class StackLevelTooDeepError < TeamCityError; end
         class NoPublicKeyError < TeamCityError; end
         class PublicKeyExpiredError < TeamCityError; end
-        class DecryptionException < TeamCityError; end
+        class DecryptionError < TeamCityError; end
         class ServerNeedsSetupError < TeamCityError; end
 
+        # Checks if the target is JetBrains TeamCity. The login module should call this.
+        #
+        # @return [Boolean] TrueClass if target is TeamCity, otherwise FalseClass
+        def check_setup
+          request_params = {
+            'method' => 'GET',
+            'uri' => normalize_uri(@uri.to_s, LOGIN_PAGE)
+          }
+          res = send_request(request_params)
+
+          if res && res.body.include?('Log in to TeamCity')
+            return true
```

Comment:
I think unfortunately the API is different from what we'd expect; I think this is

```
          if !(res && res.code == 200 && res.body&.include?('Log in to TeamCity'))
            return "Unable to locate \"Log in to TeamCity\" in headers. (Is this really TeamCity?)"

          end
          false
```